src/api/objects/ObjectAPI.js

```diff
@@ -477,7 +481,7 @@ ObjectAPI.prototype.mutate = function (domainObject, path, value) {
 
     if (this.isTransactionActive()) {
         this.transaction.add(domainObject);
-    } else {
+    } else if (!this.isSavingInprogress) {
```

Comment:
Mutations must be either added to a transaction or _immediately_ saved. 

I think that this change creates a third path there neither of those happen. The mutation _may_ accidentally get persisted by a future mutation to the object, but we can't rely on that.

What was causing the original issue?

Reply:
hmm, inside `save` we call `mutate` and inside `mutate` we call `save`, thats what causing this issue.